enhancements/windows-containers/disconnected-support.md

```diff
@@ -0,0 +1,439 @@
+* Supporting clusters that use only `ImageContentSourcePolicy` resources. These were [deprecated in 4.13](https://docs.openshift.com/container-platform/4.14/release_notes/ocp-4-14-release-notes.html#ocp-4-14-deprecated-removed-features) 
+and users [should migrate to IDMSes](https://docs.openshift.com/container-platform/4.14/openshift_images/image-configuration.html#images-configuration-registry-mirror-convert_image-configuration) instead.
+* Using [`ImageContentPolicy` resources](https://docs.openshift.com/container-platform/4.14/rest_api/config_apis/imagecontentpolicy-config-openshift-io-v1.html)
+
+
+## Proposal
+
+### Design Details
+
+There are a couple pieces of work as part of this enhacement. All changes detailed in this enhancement proposal will be
+limited to the Windows Machine Config Operator and its sub-component, Windows Instance Config Daemon (WICD).
+First, WMCO will set the `config_path` variable in the containerd config file.
+Next, the operator will generate registry config files based on cluster resources and publish them to WICD. 
+And lastly, WICD will consume the generate registry config and create them on each Windows instance.
+
+WMCO will use the mirrors data in `ImageDigestMirrorSet` and `ImageTagMirrorSet` resources 
+to create a set of [hosts.toml files](https://github.com/containerd/containerd/blob/main/docs/hosts.md#hoststoml-content-description---detail),
+one for each registry mirror, that will be transferred to the Windows instance. WMCO will handle the file content 
+creation and will publish the required info to WICD. WICD will consume the given info and create/update files on the
+Windows instances.
+
+Since WMCO is an day 2 operator (optional, not part of the OCP payload), it will pick up the mirror configuration during
+runtime regardless of whether they were set during cluster install time or at some point during the cluster's lifetime.
+
+#### Setting `config_path`
+
+WMCO will need to set the `config_path` variable in the containerd config file to tell it where to look for registry
+config settings; it is currently empty in value. We should likely set it to something within our managed directories
+(e.g. "C:\\k") to ensure they're cleaned up during node deconfiguration.
+
+#### File Content
+
+WMCO will generate the required registry config files based on the IDMS/ITMS resources in the cluster.
+The operator will react to updates to the cluster's mirror settings through a new "registry controller" which will
+watch for events on IDMS/ITMS resources.
+WMCO will publish the generated file content into the windows-services ConfigMap.
+Because the windows-services ConfigMap is immutable, publishing changes to it will require a couple steps.
+
+The file content management flow will be as such:
```

Comment:
Please add links to the code if you are getting down to that level detail. I feel you should focus on it from a higher level rather than go down to the implementation details here.

Reply:
Have you taken into account the operator crashing in the middle of any of these operations? Will it be able to recover?